### DISCUSSION TITLE
The LDES client should not request the same fragment twice

### DISCUSSION THREAD

**Discussion by rorlic** [Informatievlaanderen/VSDS-Linked-Data-Interactions#125] [Ideas]
When running the GIPOD ingest LDES test, I see that the first fragment is requested twice, probably due to the 302 redirect but this  is obviously not needed:
```
Simulator listening at http://0.0.0.0:80
GET / 200
POST /ldes application/ld+json 201
POST /ldes application/ld+json 201
POST /ldes application/ld+json 201
POST /ldes application/ld+json 201
POST /ldes application/ld+json 201
POST /alias application/json 201
GET /api/v1/ldes/mobility-hindrances 302
GET /api/v1/ldes/mobility-hindrances?generatedAtTime=2022-04-19T12:12:49.47Z 200
GET /api/v1/ldes/mobility-hindrances?generatedAtTime=2022-04-19T12:12:49.47Z 200
GET /api/v1/ldes/mobility-hindrances?generatedAtTime=2022-04-21T09:38:34.617Z 200
GET /api/v1/ldes/mobility-hindrances?generatedAtTime=2022-04-28T14:50:23.317Z 200
GET /api/v1/ldes/mobility-hindrances?generatedAtTime=2022-05-06T11:55:00.313Z 200
GET /api/v1/ldes/mobility-hindrances?generatedAtTime=2022-05-13T11:36:49.04Z 200
```

**Comment by Tomvbe** [comment]
Tom Van Besien
  [8 hours ago](https://vsdscore.slack.com/archives/C04EBSQRKT4/p1680754932448419?thread_ts=1680720407.844839&cid=C04EBSQRKT4)
Dat komt omdat we de starting node eerst moeten bepalen. We volgen dus eerst de redirects of kiezen een view uit. Wanneer we een fragment met members bereiken is de opstart klaar. Dan gebruiken we die url als startpunt om de data op te halen. Dit twee keer ophalen komt dus eigenlijk maar 1 keer voor.
:+1:
1






Ranko
  [5 hours ago](https://vsdscore.slack.com/archives/C04EBSQRKT4/p1680765157891909?thread_ts=1680720407.844839&cid=C04EBSQRKT4)
hmmm, niet helemaal akkoord:
wat slaan jullie op om een fragment niet 2x te volgen?
de fragment URL of de fragment ID?
indien je het laatste zou opslaan, weet je dat je het fragment dat wordt redirected al gezien hebt ...


Ranko
  [5 hours ago](https://vsdscore.slack.com/archives/C04EBSQRKT4/p1680765238474689?thread_ts=1680720407.844839&cid=C04EBSQRKT4)
en je hebt het fragment al gezien dus je weet of er members al dan niet inzitten (edited) 


Tom Van Besien
  [4 hours ago](https://vsdscore.slack.com/archives/C04EBSQRKT4/p1680769297642119?thread_ts=1680720407.844839&cid=C04EBSQRKT4)
De starting url bepalen is eigenlijk een lostaand component die we 'ingepluggen'  voor de 'core'. Dus eenmaal die weet wat het startpunt is, gaat die de effectieve ldes-client in gang zetten en de gevonden url meegeven. Pas dan worden zaken opgeslaan en bijgehouden etc. Daarom dat je die 'dubbele' aanroep ook maar 1 keer in het begin ziet en dan nooit meer. In theorie kan je ook gewoon direct de core met je url aanroepen als je zeker bent dat dit de url is van het fragment is en niet van de collection of view.
We zouden kunnen aanpassen dat je de gevonden data ook al meegeeft maar ik weet niet of we code willen refactoren om 1 request bij de startup uit te sparen?


Ranko
  [3 hours ago](https://vsdscore.slack.com/archives/C04EBSQRKT4/p1680772577313409?thread_ts=1680720407.844839&cid=C04EBSQRKT4)
OK, np, dan documenteren we de client persistence test zo dat het duidelijk is dat enkel het eerste fragment 2x wordt opgehaald, de andere fragmenten niet
